vault-github/run.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+BOLD='\033[0;1m'
+VAULT_ADDR=${VAULT_ADDR}
+VERSION=${VERSION}
+AUTH_PATH=${AUTH_PATH}
+GITHUB_EXTERNAL_AUTH_ID=${GITHUB_EXTERNAL_AUTH_ID}
+
+# Check if vault is installed
+if ! command -v vault &>/dev/null; then
+    printf "$${BOLD}Installing vault CLI ...\n\n"
+    # check if wget is installed
+    if ! command -v wget &>/dev/null; then
+        printf "wget is not installed. Please install wget in your image.\n"
+        exit 1
+    fi
+    # check if unzip is installed
+    if ! command -v unzip &>/dev/null; then
+        printf "unzip is not installed. Please install unzip in your image.\n"
+        exit 1
+    fi
+    # check if VERSION is latest
+    if [ "${VERSION}" = "latest" ]; then
+        INSTALL_VERSION=$(curl -s https://releases.hashicorp.com/vault/ | grep -oP '[0-9]+\.[0-9]+\.[0-9]' | tr -d '<>' | head -1)
+    else
+        INSTALL_VERSION=$VERSION
+    fi
+
+    # download vault
+    wget -q -O vault.zip https://releases.hashicorp.com/vault/$${INSTALL_VERSION}/vault_$${INSTALL_VERSION}_linux_amd64.zip
+    unzip vault.zip
+    sudo mv vault /usr/local/bin
+    rm vault.zip
+fi
+
+printf "🥳 Installation complete!\n\n"
+
+# Set up Vault token
+printf "🔑 Authenticating with Vault ...\n\n"
+echo "AUTH_PATH: $AUTH_PATH"
+echo "GITHUB_EXTERNAL_AUTH_ID: $GITHUB_EXTERNAL_AUTH_ID"
+GITHUB_TOKEN=$(coder external-auth access-token $GITHUB_EXTERNAL_AUTH_ID)
+if [ $? -ne 0 ]; then
+    printf "Authentication with Vault failed. Please check your credentials.\n"
+    exit 1
+fi
+
+export VAULT_ADDR=$VAULT_ADDR
+
+# Verify Vault address
+printf "🔎 Verifying Vault address...\n\n"
+vault status
+
+# Login to Vault to using GitHub token
+printf "🔑 Logging in to Vault ...\n\n"
+vault login -no-print -method=github -path=/$AUTH_PATH token=$GITHUB_TOKEN
+
+# Add VAULT_ADDR to shell login scripts if not already present e.g. .bashrc, .zshrc
+# This is a temporary fix and will be replaced with https://github.com/coder/coder/issues/10166
+# bash
+if [[ -f ~/.bashrc ]] && ! grep -q "VAULT_ADDR" ~/.bashrc; then
```

Comment:
These files may be initially missing, but set up by user dotfiles later. I don't have a good alternative solution, but there's a good chance this will only work on the second start or something similar. If we create the files when missing, the users dotfiles may overwrite them, so same issue (works after second start).

I can't really think of a fix other than targeting `/etc` files for the shells. But that also requires `sudo` and might have other unforeseen issues.

Reply:
Yes unfortunatly. Hopefully we get rid of them when we have `coder_env`